x-pack/plugins/security_solution/public/app/index.tsx

```diff
@@ -30,6 +31,7 @@ export const renderApp = ({
 }: RenderAppProps) => {
   render(
     <SecurityApp apolloClient={apolloClient} history={history} services={services} store={store}>
+      <UpgradeEndpointPackage />
```

Comment:
@oatkiller @paul-tavares is this the right place to do this if we want the functionality to run on any of the tabs? We'll probably want to the upgrade check to run as often as possible.

Reply:
Also @XavierM any thoughts?

Reply:
Actually having it run when a user navigates between tab seems a bit excessive. If you have ideas for limiting this so it only runs when the user initially navigates to any page on the security app that'd be helpful. That way we don't have to perform the api requests when we navigate between tabs.

Reply:
Yes, I think we can add that in the start plugin.js, and we can store it in the reducer store and then have a component reading this value to show something or not. what do you think?

Reply:
Thanks @XavierM, is this the PR? https://github.com/elastic/kibana/pull/77192/files

Reply:
@XavierM will that cause the code to run less frequently?
@jonathan-buttner Why is it important that the code runs less frequently?

Reply:
After discussing this with Rob, I think we're just going to stick with doing it when any of the plugins are started because it's more simple.